compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -684,21 +684,24 @@ ConfigurationAdminImpl::AddingService(
   const ServiceReference<cppmicroservices::service::cm::ManagedService>&
     reference)
 {
-  // Lock the configurations repository so no configuration objects can be
-  // added or removed while AddingService is processing the new service.
-  std::lock_guard<std::mutex> lk{ configurationsMutex };
- 
-  const auto pid = getPidFromServiceReference(reference);
-  if (pid.empty()) {
-    const auto bundle = reference.GetBundle();
-    const auto serviceID = std::to_string(cppmicroservices::any_cast<long>(
-      reference.GetProperty(Constants::SERVICE_ID)));
-    logger->Log(
-      SeverityLevel::LOG_WARNING,
-      "Ignoring ManagedService with ID " + serviceID + " from bundle " +
-        (bundle ? bundle.GetSymbolicName() : "Unknown") +
-        " as it does not have a service.pid or component.name property");
-    return nullptr;
+  std::string pid;
```

Comment:
Perhaps a comment here explaining the extra block to scope the lock and avoid a deadlock? Just thinking about future maintainers.